memory/src/main/scala/filodb.memory/BlockManager.scala

```diff
@@ -178,7 +181,15 @@ class PageAlignedBlockManager(val totalMemorySizeInBytes: Long,
       val num: Int = Math.ceil(memorySize / blockSizeInBytes).toInt
       stats.requestedBlocksMetric.increment(num)
 
-      if (freeBlocks.size < num) tryReclaim(num)
+      if (freeBlocks.size < num) {
+        if (bucketTime.isEmpty || onDemandTimeOrderedReclamation) {
+          tryReclaim(num)
+        } else {
+            val msg = s"Unable to allocate time ordered block(s) without forcing a reclamation: " +
+                      s"num_blocks=$num num_bytes=$memorySize freeBlocks=${freeBlocks.size}"
+            throw new IllegalStateException(msg)
```

Comment:
Is this an IllegalState ? This is the error user will see. It could mean overload of ODP queries (or ingestion) trying to use up all the headroom. I think this should be some kind of a "server overloaded" exception as opposed to illegal state.

Reply:
Is there an existing class of exceptions for this sort of thing?

Reply:
I dont think so. We may have to create one.

Reply:
What happens when a query exceeds any kind of limit? Nothing for that?

Reply:
We throw IllegalArgumentException for now.